src/client-private.ts

```diff
@@ -143,7 +232,47 @@ function onSession (this: PureCloudWebrtcSdk, session) {
       this.emit('sessionEnded', session, reason);
     });
     this.emit('sessionStarted', session);
-  });
+    // if authenitcation is jwt (meaning guest/annoyomous) and sdkType is screenshare
+    // then we are sharing our screen
+  } else if (this._guest && this._sdkType === 'screenshare' && this._jwt) {
+    log.call(this, 'debug', 'sdkType is \'screenshare\' and there is a jwt');
+    if (temporaryOutboundStream) {
+      log.call(this, 'debug', 'temporaryOutboundStream exists. Adding stream to the session and setting it to _outboundStream');
+      session.addStream(temporaryOutboundStream);
+      session._outboundStream = temporaryOutboundStream;
+      temporaryOutboundStream = null;
+    }
+
+    /* don't attach? or should we and then hide it?
+      if (session.streams.length === 1 && session.streams[0].getTracks().length > 0) {
+        attachScreenShareMedia.call(this, session.streams[0]);
+      } else {
+        session.on('peerStreamAdded', (session, stream) => {
+          attachScreenShareMedia.call(this, stream);
+        });
+      }
+    */
+
+    // always accept since we are the ones who initiated the session
+    session.accept();
+
+    session.on('terminated', (session, reason) => {
+      log.call(this, 'info', 'onSessionTerminated', { conversationId: session.conversationId, reason });
+      if (session._outboundStream) {
+        session._outboundStream.getTracks().forEach(t => t.stop());
+      }
+      this.emit('sessionEnded', session, reason);
+    });
+    this.emit('sessionStarted', session);
+
+    // if we don't have a matching combination
+  } else {
```

Comment:
So what you're missing is if the session is a screen share but this is an agent context